nats.go

```diff
@@ -1445,7 +1445,7 @@ func (nc *Conn) waitForMsgs(s *Subscription) {
 		}
 
 		// Deliver the message.
-		if m != nil && (max <= 0 || delivered <= max) {
```

Comment:
This addresses following staticcheck report `github.com/nats-io/nats/nats.go:1448:19: 'x <= 0' for unsigned values of x is the same as 'x == 0'`. Adding a `max < 0` reports that `unsigned values are never < 0` too...